test/run-tests

```diff
@@ -867,17 +899,63 @@ def main():
     # default values for config
     config = {"repositories": [], "images": [], "name": "linux-system-roles-test"}
 
+    # check for settings via env. vars
+    args.secrets = os.environ.get("TEST_HARNESS_SECRETS", args.secrets)
+    args.config = os.environ.get("TEST_HARNESS_CONFIG", args.config)
+    args.cache = os.environ.get("TEST_HARNESS_CACHE", args.cache)
+    args.inventory = os.environ.get("TEST_HARNESS_INVENTORY", args.inventory)
+    args.pull_request = os.environ.get("TEST_HARNESS_PULL_REQUEST", args.pull_request)
+    args.use_images = os.environ.get("TEST_HARNESS_USE_IMAGES", args.use_images)
+    args.dry_run = os.environ.get("TEST_HARNESS_DRY_RUN", args.dry_run)
+
     with open(args.config + "/config.json") as configfile:
         config.update(json.load(configfile))
         images = config["images"]
         random.shuffle(images)
         repos = [r.split("/") for r in config.get("repositories", [])]
         random.shuffle(repos)
 
+    # this is the name for github status reporting
+    config["name"] = os.environ.get("TEST_HARNESS_CONFIG_NAME", config["name"])
+    env_repos = os.environ.get("TEST_HARNESS_REPOSITORIES")
+    if env_repos:
+        repos = [r.split("/") for r in env_repos.split(",")]
     setup_logging(config.get("logging", {}))
 
     check_environment()
 
+    ansible_version = LooseVersion(get_ansible_version())
+    logging.info("Using Ansible version {}".format(ansible_version))
+    # this is used in PR status - context name
+    ansible_ver_str = (
+        f"ansible-{ansible_version.version[0]}.{ansible_version.version[1]}"
+    )
+
+    image_patterns = args.use_images.split(",")
+    test_images = []
+    for pattern in image_patterns:
+        for image in images:
+            min_ansible_version = image.get(
+                "min_ansible_version", DEFAULT_MIN_ANSIBLE_VERSION
```

Comment:
Why is the default "2.7" and not just "0", here?

Reply:
Makes sense to have 0 - of course the minimum supported version is 2.7 now, but hardcoding it will leave future generations wondering whether they should bump it every time an Ansible version gets out of support.

Reply:
fixed in https://github.com/linux-system-roles/test-harness/pull/87/commits/b41464cbac23050b63fb78c106256e56028731a6